R/coerce_to_messydate.R

```diff
@@ -11,90 +20,113 @@
 #' as_messydate("2021-02-01")
 #' as_messydate("20-01-2021")
 #' as_messydate("01-20-2021")
+#' as_messydate("20 September 2021")
 #' as_messydate("2021-02-01?")
 #' as_messydate("2021-02-01~")
 #' as_messydate("2021-02-01%")
 #' as_messydate("2021-02-01..2021-02-28")
 #' as_messydate("{2021-02-01,2021-02-28}")
+#' as_messydate("Two of February of two thousand twenty-two",
+#' from_text = TRUE)
```

Comment:
- [ ] More natural English would be "Second of February, two thousand and twenty-two".
- [ ] Why is the extra argument necessary? Is it too computationally expensive to search for dates otherwise
- [ ] Consider comparing approach taken here with [https://github.com/stefanieschneider/unstruwwel](https://github.com/stefanieschneider/unstruwwel)

Reply:
Thank you for the great feedback @jhollway !

I removed the extra argument and integrated the possibility of retrieving dates from text to the function itself. As for the more natural dates in texts, after a few tweaks the function also works with such formulations, I have changed the examples in documentation to reflect such.

As for expanding and converting additional (more complex) types of dates in text (i.e. negative historical, approximate, ranges, and sets) as `{unstruwwel}` does, I opened an issue for this to be expanded in the future. The obvious choice would be to rely on the `{unstruwwel}` package, but they are not on CRAN...

Please let me know what you think of the changes and if there are any other improvements you can think of. Thank you.